acasclient/lsthing.py

```diff
@@ -2076,6 +2109,62 @@ def _upload_file_values_from_state_dict(state_dict):
             return state_dict
         self.metadata = _upload_file_values_from_state_dict(self.metadata)
         self.results = _upload_file_values_from_state_dict(self.results)
+    
+    def _get_ddicts(self):
+        """Extract a unique list of DDict objects from the CodeValue attributes of a SimpleThing
+        :return: dictionary of { (code_type, code_kind, code_origin): DDict }
+        """
+        ddicts = {}
+        state_dicts = [self.metadata, self.results]
+        for state_dict in state_dicts:
+            for _, values_dict in state_dict.items():
+                for _, value in values_dict.items():
+                    if isinstance(value, CodeValue):
+                        ddict = None
+                        if value.code_origin.upper() == ACAS_DDICT:
+                            ddict = ACASDDict(value.code_type, value.code_kind)
+                        elif value.code_origin.upper() == ACAS_LSTHING:
+                            ddict = ACASLsThingDDict(value.code_type, value.code_kind)
+                        ddicts[(ddict.code_type, ddict.code_kind, ddict.code_origin.upper())] = ddict
```

Comment:
If the code_origin doesn't match ACAS_DDICT or ACAS_LSTHING, we will here raise an `AttributeError: NoneType object has no attribute ...`. Do you think we should raise a `ValueError` with custom message?

Reply:
Yep, thanks.